apps/rush-lib/src/data/test/RushConfiguration.test.ts

```diff
@@ -163,4 +177,27 @@ describe('RushConfiguration', () => {
 
     done();
   });
+
+  it('allows the temp directory to be set via environment variable', () => {
+    const expectedValue: string = path.resolve('/var/temp');
+    process.env['rush_tempDir'] = expectedValue; // tslint:disable-line:no-string-literal
+
+    const rushFilename: string = path.resolve(__dirname, 'repo', 'rush-pnpm.json');
+    const rushConfiguration: RushConfiguration = RushConfiguration.loadFromConfigurationFile(rushFilename);
+
+    assertPathProperty('commonTempFolder', rushConfiguration.commonTempFolder, expectedValue);
+    assertPathProperty('npmCacheFolder', rushConfiguration.npmCacheFolder, `${expectedValue}/npm-cache`);
+    assertPathProperty('npmTmpFolder', rushConfiguration.npmTmpFolder, `${expectedValue}/npm-tmp`);
```

Comment:
>`${expectedValue}/npm-tmp` [](start = 71, length = 26)

path.join

Reply:
This is a unit test.

---
In reply to: [177544090](https://github.com/Microsoft/web-build-tools/pull/585#discussion_r177544090) [](ancestors = 177544090)